testnet/stacks-node/src/neon_node.rs

```diff
@@ -624,35 +624,37 @@ fn spawn_peer(
                     &handler_args,
                     &mut expected_attachments,
                 ) {
-                    Ok(res) => res,
+                    Ok(network_result) => {
```

Comment:
I think we might want to mitigate this approach. If the node is being dispatching its events to an event observer, crash, if not continue?
Otherwise, the event observer could potentially miss a few things:

```rust
RelayerDirective::HandleNetResult(ref mut net_result) => {
                    debug!("Relayer: Handle network result");
                    let net_receipts = relayer
                        .process_network_result(
                            &local_peer,
                            net_result,
                            &mut sortdb,
                            &mut chainstate,
                            &mut mem_pool,
                            Some(&coord_comms),
                        )
                        .expect("BUG: failure processing network results");

                    let mempool_txs_added = net_receipts.mempool_txs_added.len();
                    if mempool_txs_added > 0 {
                        event_dispatcher.process_new_mempool_txs(net_receipts.mempool_txs_added);
                    }

                    // Dispatch retrieved attachments, if any.
                    if net_result.has_attachments() {
                        event_dispatcher.process_new_attachments(&net_result.attachments);
                    }

                    // synchronize unconfirmed tx index to p2p thread
                    send_unconfirmed_txs(&chainstate, unconfirmed_txs.clone());
                }
```